WordPress/Classes/Services/MediaCoordinator.swift

```diff
@@ -305,9 +305,23 @@ class MediaCoordinator: NSObject {
                             success: {
                                 self.end(media)
         }, failure: { error in
-            guard let nserror = error as NSError? else {
-                return
-            }
+            // Ideally the upload service should always return an error.  This may be easier to enforce
+            // if we update the service to Swift, but in the meanwhile I'm instantiating an unknown upload
+            // error whenever the service doesn't provide one.
+            //
+            let nserror = error as NSError?
+                ?? NSError(
```

Comment:
The idea is to attach this error so we can see that it's happening? Because in case of the `unknownUploadError` we don't have information about what happened, right?

Anyway, I agree with you. Is better than just ignoring it...

Reply:
Nopes, the idea is to avoid silencing errors.

Ideally this scenario should never be possible and the APIs we use should make sure an error is returned whenever there is one, but that's beyond the scope of this PR.